src/plugin.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * The current plugin state
+ */
+const state = {
```

Comment:
I assume that this is a pure js in html. If yes, those global const might be overlapping with some other plugins/main app. Might be the better to place everything into one function/class that will close everything inside it's scope.